### PR TITLE
Components: ResizableBox, add showHandle prop

### LINKED ISSUE #16652: ResizableBox example does not resize
**Describe the bug**
The ResizableBox example in README.md does not resize and cursor never changes to resize syle.

Here is the code I used:

```
import * as React from 'react';
import { ResizableBox } from '@wordpress/components';

const Edit = (props: any) => {
  const {
    attributes: { height, width },
    setAttributes,
    toggleSelection,
  } = props;

  return (
    <ResizableBox
      size={{
        height,
        width,
      }}
      style={{
        backgroundColor: 'orange',
      }}
      minHeight="50"
      minWidth="50"
      enable={{
        top: false,
        right: true,
        bottom: true,
        left: false,
        topRight: false,
        bottomRight: true,
        bottomLeft: false,
        topLeft: false,
      }}
      onResizeStop={(event, direction, elt, delta) => {
        setAttributes({
          height: parseInt(height + delta.height, 10),
          width: parseInt(width + delta.width, 10),
        });
        toggleSelection(true);
      }}
      onResizeStart={() => {
        toggleSelection(false);
      }}
    >Some text</ResizableBox>
  );
};

export default Edit;
```

But when I use original module instead of exported one, it works as expected:

```
import { Resizable as ResizableBox } from 're-resizable';
```

Here is what ResizableBox from components renders onto page:
```
<div data-block="a4765993-8e9b-4331-bc85-78ed55288e4f"><div class="components-resizable-box__container" style="position: relative; user-select: auto; background-color: orange; width: 100px; height: 100px; box-sizing: border-box;">Some text<span><div class="components-resizable-box__handle components-resizable-box__side-handle components-resizable-box__handle-right" style="position: absolute; user-select: none; cursor: col-resize;"></div><div class="components-resizable-box__handle components-resizable-box__side-handle components-resizable-box__handle-bottom" style="position: absolute; user-select: none; cursor: row-resize;"></div><div class="components-resizable-box__handle components-resizable-box__corner-handle components-resizable-box__handle-bottom components-resizable-box__handle-right" style="position: absolute; user-select: none; cursor: se-resize;"></div></span></div><div class="__resizable_base__" style="width: 100%; height: 100%; position: absolute; transform: scale(0, 0); left: 0px; flex: 0 1 0%;"></div></div>
```

Here what original Resizable component from 're-resizable' module renders onto page:
```
<div data-block="f51637e3-cc3f-492a-8eb5-c456ea316893"><div style="position: relative; user-select: auto; background-color: orange; width: 100px; height: 100px; box-sizing: border-box;">Some text<span><div class="" style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"></div><div class="" style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"></div><div class="" style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; bottom: -10px; cursor: se-resize;"></div></span></div><div class="__resizable_base__" style="width: 100%; height: 100%; position: absolute; transform: scale(0, 0); left: 0px; flex: 0 1 0%;"></div></div>
```
Note: Gutenberg Handbook also uses same example.

### PR DESCRIPTION
**Pull request by ItsJonQ**
Screen Capture on 2019-10-24 at 18-16-57

## Description

This update adds a new prop to the ResizableBox component. This new prop `showHandle` (`bool`) determines if the draggable resize handles should be visible or not. This update provides more flexible for how `ResizableBox` can render.

## How has this been tested?
Storybook

## Screenshots

Screen Shot 2019-10-24 at 6 18 22 PM

## Types of changes

* Add property (with default value) to `ResizableBox`
* Add  accompanying CSS className + style for that prop
* Add Storybook example + knobs 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 

Helps resolve https://github.com/WordPress/gutenberg/issues/16652

### CODE REVIEW

File: packages/components/src/resizable-box/style.scss [hunk 1]
```diff
@@ -9,6 +9,10 @@
 	.components-resizable-box__container.is-selected & {
 		display: block;
 	}
+	// Show the resize handle if set in props
```
Comment (gziolo) [inline packages/components/src/resizable-box/style.scss]: Can you combine those 2 rules which set `display:block` into one? They seem to have the same purpose.

File: packages/components/src/resizable-box/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+export default { title: 'ResizableBox', component: ResizableBox };
+
+const Example = ( props ) => {
+	const [ attributes, setAttributes ] = useState( { height: 200, width: 400 } );
+	const { height, width } = attributes;
+	const { content, ...restProps } = props;
+
+	return (
+		<div style={ { margin: 30 } }>
```
Comment (gziolo) [inline packages/components/src/resizable-box/stories/index.js]: What do you think about refactoring Example to be `ResizableBoxWithState` to return only this part:
```js
<ResizableBox
	{ ...restProps }
	size={ {
		height,
		width,
	} }
	onResizeStop={ ( event, direction, elt, delta ) => {
		setAttributes( {
			height: parseInt( height + delta.height, 10 ),
			width: parseInt( width + delta.width, 10 ),
		} );
	} }
>
	{ children }
</ResizableBox>
```
and moving unrelated styling to the parent component?

This way we could see more code in the preview in Storybook.

File: packages/components/src/resizable-box/index.js [hunk 3]
```diff
@@ -23,7 +23,8 @@ function ResizableBox( { className, ...props } ) {
 		<Resizable
 			className={ classnames(
 				'components-resizable-box__container',
-				className,
+				showHandle && 'is-show-handle',
```
Comment (gziolo) [inline packages/components/src/resizable-box/index.js]: Did you consider an alternative name for `is-show-handle`? It doesn't read great. `shows-handle` would be better but it probably doesn't follow the naming convention for the class names. Maybe `has-resize-handle` based on the code comment?

Comment (ItsJonQ) [reply packages/components/src/resizable-box/index.js]: Ah yes! `has-resize-handle` reads better. It's a struggle sometimes with these names 😂

File: packages/components/src/resizable-box/style.scss (hunk 1 above)
Comment (ItsJonQ) [reply packages/components/src/resizable-box/style.scss]: Sure thing!

**Comment by ItsJonQ** [comment]
@gziolo Thank you for those suggestions! I just made the updates ❤️

File: packages/components/src/resizable-box/README.md [hunk 4]
```diff
@@ -53,3 +53,11 @@ const Edit = ( props ) => {
 	);
 }
 ```
+
+### Props
```
Comment (gziolo) [inline packages/components/src/resizable-box/README.md]: By the way, this is why I like when you abstract away external code, you get the freedom to tailor it to your needs 👍

**Review by gziolo — APPROVED** [review]
I got super excited about testing this PR. Storybook is amazing, it makes the testing a real pleasure! 💯 

Great work on this one, thanks for addressing my feedback. This is ready to go. 

I think this PR will auto-close the linked issues in the description, so you should double-check if that is expected. Sometimes you need to be very careful because those trigger words recognized by GitHub don't look at the full context :) You wrote `helps resolve XYZ` but GitHub understood only `resolve XYZ` 😅

File: packages/components/src/resizable-box/README.md (hunk 4 above)
Comment (ItsJonQ) [reply packages/components/src/resizable-box/README.md]: Yay! Thank you 🙌

**Comment by ItsJonQ** [comment]
> Storybook is amazing, it makes the testing a real pleasure!
@gziolo You are very welcome!! It's very exciting stuff. And makes it much easier for new comers like myself to work with and focus just on the UI 😊 

> You wrote helps resolve XYZ but GitHub understood only resolve XYZ

Oh gotcha! Also... TIL `resolves` is a trigger word 😮

**Comment by ItsJonQ** [comment]
> I feel like having it comment on every build would just add a lot of noise in PRs and ping everyone unnecessarily.

Gotcha! I believe it only adds one Github comment. Still, you make a good point. If you think it'll be too noisy, we don't have to add

**Pull request merged**